cardano-crypto-class/src/Cardano/Crypto/Libsodium/UnsafeC.hs

```diff
@@ -0,0 +1,10 @@
+{-# LANGUAGE CApiFFI             #-}
+module Cardano.Crypto.Libsodium.UnsafeC (
+    c_sodium_compare_unsafe,
+    ) where
+
+import Foreign.C.Types (CSize (..))
+import Foreign.Ptr (Ptr)
+
+-- | Unsafe variant of 'c_sodium_compare'.
```

Comment:
Would be nice to be clear on how exactly this is unsafe!

Reply:
foreign import capi *unsafe*

Reply:
I think @Oleg would be the person to ask, but I would wager that the `import unsafe` part is what this is about. In a nutshell, the "safe" version (found in cardano-crypto-class/src/Cardano/Crypto/Libsodium/C.hs) is a "safe" import, so that suggests that the "safe" version may call back into Haskell, while the "unsafe" version may not (but will produce better performance, because it doesn't have to do the associated bookkeeping). Why this is necessary however, I don't know.